lib/ansible/modules/cloud/amazon/redshift_iam_roles.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+module: redshift_iam_roles
+short_description: This module adds/removes IAM roles to/from Redshift clusters.
+version_added: 2.6
+author: Aaron Smith (@slapula)
+description:
+    - "This module adds or removes a list of IAM roles to a give Redshift Cluster"
+options:
+    cluster:
+        description:
+            - "This is the cluster you'd like to modify"
+        required: true
+    state:
+        description:
+            - "Desired state of the list of IAM roles on the give Redshift cluster"
+        required: true
+        choices: ['present', 'absent']
+    roles:
+        description:
+            - "This is a list of IAM roles (Full ARN) to add to the Redshift cluster. Roles attached to the cluster that are not on the list will be removed."
+        required: true
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+EXAMPLES = '''
+- name: add IAM roles to a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: present
+    roles:
+        - "arn:aws:iam::123456789012:role/superuser"
+        - "arn:aws:iam::123456789012:role/new_hotness"
+
+- name: remove all IAM roles from a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: absent
```

Comment:
I think it might be safer if roles had to explicitly be an empty list to remove all of them.
```
- name: remove all IAM roles from a redshift cluster
  redshift_iam_roles:
    cluster: "staging"
    state: absent
    roles: []
```